lib/active_model/serializer/attributes.rb

```diff
@@ -0,0 +1,59 @@
+module ActiveModel
+  class Serializer
+    # Defines the attributes to be serialized.
+    #
+    module Attributes
+      extend ActiveSupport::Concern
+
+      included do |base|
+        base.class_attribute :_attributes
+        base._attributes ||= []
+        base.class_attribute :_attributes_keys
+        base._attributes_keys ||= {}
+      end
+
+      module ClassMethods
+        def inherit_attributes(base)
+          base._attributes = _attributes.dup
+          base._attributes_keys = _attributes_keys.dup
+        end
+
+        def attributes(*attrs)
+          attrs = attrs.first if attrs.first.class == Array
+
+          attrs.each do |attr|
+            attribute(attr)
+          end
+        end
+
+        def attribute(attr, options = {}, &block)
+          key = options.fetch(:key, attr)
+          _attributes_keys[attr] = { key: key } if key != attr
+          _attributes << key unless _attributes.include?(key)
+
+          ActiveModelSerializers.silence_warnings do
+            define_method key do
```

Comment:
this absolutely needs to be document above the method signature with some examples. 
I am a HUUUGE fan of metaprogramming, but it can be a pain in the butt if people aren't expecting it. So, having some example syntax in a method yardoc would be fabulous.